geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -515,17 +510,15 @@ protected boolean processHandShake() {
         }
         if (numRefs != null) {
           if (acceptHandShake(endpointType, queueSize)) {
-            numRefs.incr();
+            ++numRefs;
```

Comment:
The old code updated a Counter held in the map.  This is just incrementing an Integer that you plucked out of the map.  It isn't going to update the value held in the map.

Reply:
Since this is using an Integer rather than an int, it's actually passing around a reference to the boxed value.  I was a little unsure about this myself, but verified in the debugger.